torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1443,7 +1444,19 @@ def coerce_runtime_tangent_tracing_memory_format(x, memory_format):
             return x
 
         is_subclass: bool = is_traceable_wrapper_subclass(x)
-        mem_format = memory_format[0] if is_subclass else memory_format
+        mem_format = memory_format
+        if is_subclass:
+            memory_format_for_dense_tensor = not isinstance(memory_format, list)
+            if isinstance(x, AsyncCollectiveTensor) and memory_format_for_dense_tensor:
+                # This is AsyncCollectiveTensor, that we have not seen during tracing time.
+                while True:
```

Comment:
hmm, I don't think it's possible to end up with an `AsyncCollectiveTensor(AsyncCollectiveTensor(...)` (that would probably a bug with the subclass itself). So I'd rather just not have a while loop here. Is there a place where you saw this show up?